docs/how-to-guides/contribute.md

```diff
@@ -0,0 +1,41 @@
+# How to contribute
+
+## Overview
+
+This document explains the processes and practices recommended for contributing enhancements to the GitHub Runner operator.
+
+* Generally, before developing enhancements to this charm, you should consider [opening an issue](https://github.com/canonical/github-runner-operator/issues) explaining your use case.
+* If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
+* Familiarizing yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+* All enhancements require review before being merged. Code review typically examines
+  * code quality
+  * test coverage
+  * user experience for Juju administrators of this charm.
+For more details, check our [contributing guide](https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md).
+
+## Developing
+
+For any problems with this charm, please [report bugs here](https://github.com/canonical/github-runner-operator/issues).
+
+The code for this charm can be downloaded as follows:
+
+```
+git clone https://github.com/canonical/github-runner-operator.git
+```
+
+To run tests, run `tox` from within the charm code directory.
+
+To build and deploy a local version of the charm, simply run:
+
+```
+charmcraft pack
+# Ensure you're connected to a juju k8s model
+# Configure the machine resource created by the model
+juju set-model-constraints mem=8G cores=2 root-disk=50G
```

Comment:
How does this tie with the default options for memory and disk usage in `config.yaml`, any relation?

Reply:
These constraints dictates the amount of resource juju allocate for the juju machine created.
The charm configuration for memory and disk is the size of the LXD VM spawned for the runner.

The juju machine would need to have enough resource for the charm configuration to work. The juju machine constraint also needs to factor in the resource used by the juju machine itself.

Generally, it is a good idea to allocate 20GiB disk for the juju machine as juju log can consume disk space quickly.  
Generally, I allocate something like 8GiB memory for the juju machine.

Reply:
Minor comments